src/test/java/io/supertokens/test/userRoles/UserRolesTest.java

```diff
@@ -231,4 +233,59 @@ public void testCreatingARoleWithPermissionsAndAddingOneMorePermission() throws
         assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
     }
 
+    @Test
+    public void testAddRoleToUserResponses() throws Exception {
+        String[] args = { "../" };
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) {
+            return;
+        }
+        String[] roles = new String[] { "role" };
+        String userId = "userId";
+        UserRolesSQLStorage storage = StorageLayer.getUserRolesStorage(process.main);
+
+        // assign an unknown role to a user, it should throw UNKNOWN_ROLE_EXCEPTION
+        {
+            Exception error = null;
+            try {
+
+                UserRoles.addRoleToUser(process.main, userId, "unknown_role");
+            } catch (Exception e) {
+                error = e;
+            }
+
+            assertNotNull(error);
+            assertTrue(error instanceof UnknownRoleException);
+        }
+
+        // create a role and assign the role to a user, check that the user has the role
+        {
+            // create role
+            UserRoles.createNewRoleOrModifyItsPermissions(process.main, roles[0], null);
+            // assign role to user
+            UserRoles.addRoleToUser(process.main, userId, roles[0]);
+
+            // check that user actually has the role
+            String[] userRoles = storage.getRolesForUser(userId);
+            Utils.checkThatArraysAreEqual(roles, userRoles);
+        }
+
+        // assign the same role to the user again, it should not throw an exception
+        {
+            // assign the role to the user again
+            UserRoles.addRoleToUser(process.main, userId, roles[0]);
+
+            // check that the user still has the same role/ no additional role has been added
+            String[] userRoles = storage.getRolesForUser(userId);
+            Utils.checkThatArraysAreEqual(roles, userRoles);
+
+        }
+
```

Comment:
Assign another role to the user and then check that there are two roles for the user